scalyr_agent/sharded_copying_manager/copying_manager.py

```diff
@@ -1137,20 +1192,66 @@ def _scan_for_new_logs_if_necessary(
             log_matchers, checkpoints=checkpoints, copy_at_index_zero=copy_at_index_zero
         )
 
+    def __prepare_checkpoints(self):
+        """
+        Combines all worker checkpoints that were left from the previous agent run.
+        It combines checkpoint files into one single file in order to
+        allow workers to rewrite the content of this files without losing any data.
+        """
+
+        checkpoints_dir_path = os.path.join(
+            self.__config.agent_data_path, "checkpoints"
+        )
+
+        # if there is a file called as the checkpoints folder, remove it
+        if os.path.isfile(checkpoints_dir_path):
+            os.unlink(checkpoints_dir_path)
+
+        # create new checkpoint folder if does not exist.
+        if not os.path.exists(checkpoints_dir_path):
```

Comment:
Are there cases where this would happen?  In general, our install scripts (I believe) create the checkpoint directories.  So, we shouldn't have files with their names.  Or, am I missing something and you are trying to address something here?

Reply:
Yes, I guess it should be better to create it on installation. I've modified the `build_packages.py` file for that. Thanks for the good point.